pkg/identityserver/api_key_utils.go

```diff
@@ -29,8 +31,22 @@ var apiKeyHashSettings auth.HashValidator = pbkdf2.PBKDF2{
 	SaltLength: 16,
 }
 
+var errExpiryDateInPast = errors.DefineInvalidArgument("expiry_date_invalid", "expiry date is in the past")
+var errInvalidDateFormat = errors.DefineInvalidArgument("expiry_date_format_invalid", "invalid expiry date format (use YYYY-MM-DD)")
+
+func parseAndValidateAPIKeyExpiry(expiry string) (*time.Time, error) {
+	expiryDate, err := time.Parse("2006-01-02", expiry)
+	if err != nil {
+		return nil, errInvalidDateFormat
+	}
+	if expiryDate.Before(time.Now()) {
+		return nil, errExpiryDateInPast
+	}
+	return &expiryDate, nil
+}
```

Comment:
This needs to be moved to the frontends (in this case, the Go code should go to the CLI).